### ISSUE TITLE
New privacy switches in macOS 15 are not managed by `allowDiagnosticSubmission`

### ISSUE THREAD

**Issue by qupig** [usnistgov/macos_security#434]
### Problem to solve

Two new data collection switches in macOS 15 (enabled by default):
- Analytics & Improvements -> Improve Assistive Voice Features
- Spotlight -> Improve Search

The restrictions key `allowDiagnosticSubmission` cannot cover the newly added switches, as shown in the screenshots:

![Improve Assistive Voice Features](https://github.com/user-attachments/assets/34420cd0-9c02-4e65-8bdb-456f325e0f5b)
![Spotlight Improve Search](https://github.com/user-attachments/assets/6ef18584-2b4d-4581-8df3-f109949e914d)

### Documentation

https://developer.apple.com/documentation/devicemanagement/restrictions

### Links / references

https://support.apple.com/en-me/guide/mac-help/mchl54d95e8a/mac
https://www.apple.com/legal/privacy/data/en/improve-assistive-voice-features/

**Comment by vaughnhart** [comment]
This is a find! with a capital FIND! I wonder if these "features" bypass other restrictions with a nuisance. They're not exactly Siri, and Spotlight. I wonder what the underlying "linux" subsystem is?

**Comment by robertgendler** [comment]
I think we found those and added rules for them.
https://github.com/usnistgov/macos_security/blob/main/rules/system_settings/system_settings_improve_assistive_voice_disable.yaml

https://github.com/usnistgov/macos_security/blob/main/rules/system_settings/system_settings_improve_search_disable.yaml

**Comment by robertgendler** [comment]
@vaughnhart there is no linux subsystem. It isn't linux. It isn't anywhere near linux. macOS is a unix and linux is not a unix.

**Comment by vaughnhart** [comment]
yeppers... private... I was more thinking about how spotlight was feeding from, if I remember correctly, lookup and another search database that could be controlled on the command line. I wondered if the same was for these two "options" as well.

**Comment by robertgendler** [comment]
Good point! We'll try to remember to update the discussion and/or title a bit.

**Comment by JamminR** [comment]
> I think we found those and added rules for them. https://github.com/usnistgov/macos_security/blob/main/rules/system_settings/system_settings_improve_assistive_voice_disable.yaml
> 
> https://github.com/usnistgov/macos_security/blob/main/rules/system_settings/system_settings_improve_search_disable.yaml

Though my org using this setting for "improve assistive voice" shows setting as disabled, the setting is not greyed out and the slider can be moved to 'on' position but immediately moves back to off.
Is this a concern, or with the setting method it was disabled, is that 'normal' and it will always remain disabled even though not greyed out.
<img width="467" alt="image" src="https://github.com/user-attachments/assets/dc055ee6-b10d-495c-87c6-e907cfdc6bee" />

**Comment by robertgendler** [comment]
@JamminR  That's expected when it's not a setting that Apple officially supports as being manageable. Nobody has programmed it to "lock". But if you re open system settings it's off.

**Comment by JamminR** [comment]
Thank you for the response. I'll stop looking for a solution to a non-problem, and will go back to occasional Apple managed device github settings review in hopes they add ability to lock it as they have the other settings.
